bundles/org.openhab.binding.omnilink/src/main/java/org/openhab/binding/omnilink/internal/handler/AudioSourceHandler.java

```diff
@@ -0,0 +1,160 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.omnilink.internal.handler;
+
+import static org.openhab.binding.omnilink.internal.OmnilinkBindingConstants.*;
+
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.StringType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.RefreshType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.digitaldan.jomnilinkII.Message;
+import com.digitaldan.jomnilinkII.MessageTypes.AudioSourceStatus;
+import com.digitaldan.jomnilinkII.OmniInvalidResponseException;
+import com.digitaldan.jomnilinkII.OmniUnknownMessageTypeException;
+
+/**
+ * The {@link AudioSourceHandler} defines some methods that are used to
+ * interface with an OmniLink Audio Source. This by extension also defines the
+ * Audio Source thing that openHAB will be able to pick up and interface with.
+ *
+ * @author Brian O'Connell - Initial contribution
+ */
+@NonNullByDefault
+public class AudioSourceHandler extends AbstractOmnilinkHandler {
+    private final Logger logger = LoggerFactory.getLogger(AudioSourceHandler.class);
+    private static final long POLL_DELAY = 5; // 5 Second polling
+    private @Nullable ScheduledFuture<?> scheduledPolling = null;
+
+    public AudioSourceHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void initialize() {
+        cancelPolling();
+        boolean autoStart = ((Boolean) getThing().getConfiguration().get(THING_PROPERTIES_AUTO_START)).booleanValue();
+        int sourceNumber = getThingNumber();
+
+        if (autoStart) {
+            logger.debug("Autostart enabled, scheduling polling for Audio Source {}", sourceNumber);
+            schedulePolling();
+        } else {
+            logger.debug("Autostart disabled, not scheduling polling for Audio Source {}", sourceNumber);
+        }
+        updateStatus(ThingStatus.ONLINE);
+    }
+
+    @Override
+    public synchronized void dispose() {
+        cancelPolling();
+        super.dispose();
+    }
+
+    private synchronized void cancelPolling() {
+        final ScheduledFuture<?> scheduledPolling = this.scheduledPolling;
+        if (scheduledPolling != null && !scheduledPolling.isDone()) {
```

Comment:
No harm in canceling a task that is already cancelled.
```suggestion
        if (scheduledPolling != null) {
```